src/System.Buffers/src/System/Buffers/BufferData.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace System.Buffers
+{
+    public struct BufferData
+    {
+        internal byte[] _data;
+        internal bool _allocd;
```

Comment:
That's a good idea, but what about when all the buffers are returned? Then we would have extra buffers with no space to put them...re-allocing the array for more space would be bad and adding special code to handle that case inside the spin lock seems like it would hurt perf.

Thoughts?